### PR TITLE
Added annotations.md and reworked sdk.md

### PR DESCRIPTION
**Pull request by juanjux**
- Added draft doc about annotating the AST
- Improved annotations document
- Reworked sdk.md, review, fixed internal links
- Grammar fix

### CODE REVIEW

File: driver/annotations.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+# Adding UAST Annotations
+
+# Intro
+
+Once you have written the [native part of a programming language's
```
Comment (smola) [inline driver/annotations.md]: `s/native part of a programming language's driver/AST parser/` ?

File: driver/annotations.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+driver](driver/sdk.md) the
+next step is to write the annotation code. This is the Go language code that will
+establish the rules to transform the native driver AST into the UAST and the
```
Comment (smola) [inline driver/annotations.md]: `s/the native driver AST/the original AST` ?

File: driver/annotations.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+implementation uses two objects, the ObjectToNoder and the AnnotationRules on two
+separate files.  Most of the files related to the Go part of the driver are
+auto-generated by the [`bblfsh-sdk init`
+tool](driver/sdk.md), but in
+order to translate the native AST to the UAST you need to complete two skeleton
+files that are referenced in the (auto-generated) driver/main.go file.
+
+## File: parser.go
+
+### ObjectToNoder struct
```
Comment (smola) [inline driver/annotations.md]: This would be actually the `ToNoder`. `ObjectToNoder` is just a possible implementation.

File: driver/annotations.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+This is a struct that you'll need to fill in the `driver/normalizer/parser.go`
+file. It serves to establish the role of some internal keys in the native AST into
+specific key roles in the UAST. This is needed so the rule applying process knows
+what parts of the native AST use to do the translation. The skeleton of this
+struct will be automatically generated by the `bblfsh-sdk init` command. 
+
+The meaning of the fields in the ObjectToNoder struct are:
+
+- InternalTypeKey (string): this is the name of the key that the native AST uses
```
Comment (smola) [inline driver/annotations.md]: Maybe all of this should go to `ObjectToNoder` godoc and then just link to it from here.

File: driver/annotations.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+  ```
+
+  In this case, the new nodes will have the InternalTypeKey "If.body", "If.orelse"
+  and "If.condition" and with these names you should be able to write specific 
+  matching rules in the annotation.go file.
+
+### UASTParserBuilder function
+
+This automatically generated function will return a driver.UASTParser object that
+will be constructed using your provided ObjectToNoder struct. This is the function
```
Comment (smola) [inline driver/annotations.md]: You have to provide a `ToNoder`, and `ObjectToNoder` is one of the possible implementations you could use.

File: driver/sdk.md [hunk 6]
```diff
@@ -1,39 +1,43 @@
...
+required to create a Babelfish driver and some APIs to analyze or transform
+the generated UAST.
+
+## Preparations
 
-## Prerequisites
+### Dependencies
 
-Babelfish has the following prerequisites:
+Babelfish has the following dependencies:
```
Comment (smola) [inline driver/sdk.md]: `s/Babelfish/Babelfish SDK/`

File: driver/sdk.md [hunk 7]
```diff
@@ -76,78 +84,267 @@ $ bblfsh-sdk update
...
+one language feature, or even several if the feature is complex. This will make
+the tests more atomic and help immensely with your sanity when trying to debug
+failed integration tests.
+
+### Officially adding your driver to Babelfish
+
+The last step if to have your language added to the [Babelfish Github's
+Project](https://github.com/bblfsh) so more people can use/break/debug or improve
+it.  For this you need to request a new repository to be created which you can
+make by opening a PR or talking with us in the [public Slack
```
Comment (smola) [inline driver/sdk.md]: There no place to create a PR for a new driver. Maybe we could tell people to open an issue in the server?

File: driver/sdk.md [hunk 8]
```diff
@@ -76,78 +84,267 @@ $ bblfsh-sdk update
...
+
+The fields in the JSON requests and responses are detailed on the [internal
+protocol](driver/internal-protocol.md)
+page.
+
+*Note:* When extracting the native AST you should try to keep as much information
+as possible following the guidelines explained in the ["From Code to
+AST"](uast/code-to-ast.html) page.
+
+#### Native parser unittests
```
Comment (smola) [inline driver/sdk.md]: `s/unittests/unit tests/`

File: driver/sdk.md [hunk 9]
```diff
@@ -76,78 +84,267 @@ $ bblfsh-sdk update
...
 
-The AST to UAST normalization is written in Go. The main file to be edited is
-`driver/normalizer/normalizer.go`.
+Once you've created your unit tests, you must modify the Makefile in the root
+of the driver filesystem adding the commands to run them under the Makefile
+target `test-parser-internal`. For example for a Python driver using the 
+Python default test framework it could look like:
 
-TODO
+test-parser-internal:
```
Comment (smola) [inline driver/sdk.md]: Use a markdown block here.

File: driver/sdk.md [hunk 10]
```diff
@@ -76,78 +84,267 @@ $ bblfsh-sdk update
...
+    adduser ${BUILD_USER} -u ${BUILD_UID} -D -h /opt/driver/src
+RUN apk add --no-cache make git curl ca-certificates
+WORKDIR /opt/driver/src
+```
+
+As the comments in the file say, Alpine is the preferred distribution for 
+space reasons but you can use Debian slim or other slim versions if for any 
+reason Alpine don't work for your driver.
+
+You should add any packages you need to the "apk add" line. Avoid adding more
```
Comment (smola) [inline driver/sdk.md]: `s/"apk add"/\`apk add\`/`

File: driver/sdk.md [hunk 11]
```diff
@@ -76,78 +84,267 @@ $ bblfsh-sdk update
...
+lines with more `RUN apk add` commands instead because that would create more
```
Comment (smola) [inline driver/sdk.md]: `s/intermediate images/intermediate layers/`

File: driver/annotations.md (hunk 3 above)
Comment (juanjux) [inline driver/annotations.md]: I reworded that paragraph a little to explain that ToNoder is the Interface and ObjectToNoder the implementation for trees represented as JSON objects.

File: driver/annotations.md (hunk 4 above)
Comment (juanjux) [inline driver/annotations.md]: Mmmm I think that the more concise descriptions are better for an API doc as they're and these (more developed) are better on the book. No?

File: driver/annotations.md (hunk 5 above)
Comment (juanjux) [inline driver/annotations.md]: Fixed.

**Comment by juanjux** [comment]
@smola fixes applied (except for the one I comment that I think is better to left the doc more verbose and the API doc shorter).

File: driver/annotations.md [hunk 12]
```diff
@@ -0,0 +1,225 @@
+# Adding UAST Annotations
+
+# Intro
+
+Once you have written the [AST parser](driver/sdk.md) the
+next step is to write the annotation code. This is the Go language code that will
+establish the rules to transform the original AST into the UAST and the
+implementation uses two objects, the ObjectToNoder and the AnnotationRules on two
```
Comment (alcortesm) [inline driver/annotations.md]: I would say that the implementation uses two *types* instead of two objects.

I'd also reference the types by their full name: -
- `github.com/bblfsh/sdk/protocol/native.ObjectToNoder`
- `github.com/bblfsh/sdk/uast/ann.Rule`
and maybe with a link to theirs godocs.

File: driver/annotations.md [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+
+ToNoder is a interface that annotators would need to fill in the
+`driver/normalizer/parser.go` file, but a default ObjectToNoder implementation is
+provided in the SDK for trees represented as nested JSON objects. It serves to
+establish the role of some internal keys in the native AST into specific key roles
+in the UAST. This is needed so the rule applying process knows what parts of the
+native AST use to do the translation. The skeleton of this struct will be
+automatically generated by the `bblfsh-sdk init` command. 
+
+The meaning of the fields in the ObjectToNoder struct are:
```
Comment (alcortesm) [inline driver/annotations.md]: This should not be explained here, this should be explained in the package documentation and referenced here.

File: driver/annotations.md (hunk 12 above)
Comment (juanjux) [inline driver/annotations.md]: Will do, thanks.

File: driver/annotations.md (hunk 13 above)
Comment (juanjux) [inline driver/annotations.md]: Point taken (@smola said the same) will do that, thanks to both.

File: driver/annotations.md (hunk 12 above)
Comment (smola) [inline driver/annotations.md]: FYI: sdk godocs: https://godoc.org/github.com/bblfsh/sdk

Comment (juanjux) [inline driver/annotations.md]: Done

File: driver/annotations.md [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+
+The type and meaning of the fields in the native.ObjectToNoder struct can be
+checked on the [GoDoc documentation of the
+type](https://godoc.org/github.com/bblfsh/sdk/protocol/native#ObjectToNoder).
+
+### UASTParserBuilder function
+
+This automatically generated function will return a driver.UASTParser object that
+will be constructed using your provided ToNoder implementation. This is the
+function referred in the main.go file and you usually will not need to change it.
```
Comment (abeaumont) [inline driver/annotations.md]: main.go -> `main.go`, or link to the file

File: driver/annotations.md [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+
+#### Predicates
+
+Predicates are any function that follow the signature: 
+
+```go
+func(n *uast.Node) bool
+```
+
+Where uast.Node are the Nodes that will be iterated during the annotation
```
Comment (abeaumont) [inline driver/annotations.md]: Where uast.Node are the Nodes -> `n` is a pointer to the tree

File: driver/annotations.md [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+process. You could define your own predicate functions but the ann.go file
```
Comment (abeaumont) [inline driver/annotations.md]: If the normal usage doesn't need the user to build its own predicate, it shouldn't be the first thing they read. I'd rephrase:

`ann.go` fail has a list of functions with the predicates you'll probably need:
- `Any()`
- ...

These predicate functions should be enough for most cases, but you can still define your own if needed.

File: driver/annotations.md [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+has already defined the ones that you'll probably need most of the time:
+
+- `Any()`: always returns true. It's usually used to match the first node in 
+  the native AST and add rules from there.
+
+- `Not/And/Or(predicate)`: will apply the expected boolean operations to 
+  the predicate(s) used as argument.
+
+- `HasInternalType(type string)`: checks if the InternalType of the node matches
```
Comment (abeaumont) [inline driver/annotations.md]: Be consistent with the tense, either use future or present, but not a mix.

File: driver/annotations.md [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+  string. What key of the node is a Token is defined in the 
+  `parser.ObjectToNoder.TokenKeys` map.
+
+- `HasChild(predicate)`: will check if any of the first level children of
+  the current node matches the given predicate.
+
+- `HasProperty(key, value string)`: checks if the AST node has some specific key
+  and the key as the given value.
+
+- `HasInternalRole(key string)`: this is actually a very handy shortcut to
```
Comment (abeaumont) [inline driver/annotations.md]: I'd explain what it does first (as with the rest of predicates), since it's the most important point. It's relation with HasProperty is just an implementation detail, and may change in the future, I wouldn't mention it.

File: driver/annotations.md [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+  and the key as the given value.
+
+- `HasInternalRole(key string)`: this is actually a very handy shortcut to 
+  `HasProperty(uast.InternalRoleKey, value)` that will check if the node has
+  an InternalRole matching the specified key. InternalRoles are
+  added to children of any object keys holding them.
+
+#### Actions
+
+- `Roles(roles... uast.Role)`: Will add the given UAST role to the currently
```
Comment (abeaumont) [inline driver/annotations.md]: Either capitalize all the definitions or none.

File: driver/annotations.md [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+[//]: # different languages have mapped their internal roles to the UAST ones.
+
+You can also ask any doubt on the project's public [Babelfish Slack
+channel](https://sourced-community.slack.com/join/shared_invite/MTkwNTM0ODEyODIzLTE0OTYxMzc5NTMtODRhMDYyNzAyYQ)
+which is very friendly to newcomers to the project.
+
+Finally, if you really think that there isn't a correspondence in the UAST roles
+for the native role that you want to map, you can fork the [Babelfish SDK project
+on Github](https://github.com/bblfsh/sdk), add the new role to the file
+`uast/uast.go` and make a PR. Don't expect the role to be added inmediatedly;
```
Comment (abeaumont) [inline driver/annotations.md]: Not sure this is the best way to ask for UAST normalization problem feedback. Maybe opening an issue to check whether there's a real need and discuss which may be the best solution (which may include a new BIP, for example), before asking people to actually write any code is a better way.

File: driver/sdk.md [hunk 21]
```diff
@@ -1,39 +1,43 @@
...
 
-Babelfish has the following prerequisites:
+The Babelfish SDK has the following dependencies:
 
 * [Docker](https://www.docker.com/get-docker)
 * [Go](https://golang.org/dl/)
 
-## Installing
+Make sure that you've correctly set your GOROOT and GOPATH environment
+variables.
```
Comment (abeaumont) [inline driver/sdk.md]: A link to Go's documentation explaining how to do so may be handy here.

File: driver/sdk.md [hunk 22]
```diff
@@ -1,39 +1,43 @@
...
 
-## Installing
+Make sure that you've correctly set your GOROOT and GOPATH environment
+variables. 
 
-Install the Babelfish SDK:
+### Installing
+
+Install the Babelfish SDK. Native drivers and the external tools can be developed
+on any programming language but the non-native driver is developed using Go so you
```
Comment (abeaumont) [inline driver/sdk.md]: - developed *in* any language.
- non-native driver is developed using Go -> normalizer is written in Go

I find this whole paragraph confusing though. The fact that the SDK is written in go is the one that determines that you need to use go to install it, no the fact that the normalizer will have to be written in Go. I'd remove it.

File: driver/sdk.md [hunk 23]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
 
-## Creating the AST parser
+`bblfsh-sdk` doesn't update the SDK itself so if you want to use a new version of 
+the SDK you'll also have to update the Go package **first** with:
+
+```bash
+go get -u github.com/bblfsh/sdk/...
+```
+
+If the update gives you some problem you can try to delete the
```
Comment (abeaumont) [inline driver/sdk.md]: s/some/any/

File: driver/sdk.md [hunk 24]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+again to get a fresh copy.
+
+## Implementing the Driver
+
+A driver has several subcomponents. On this document and the next one
+we'll see how to implement all these parts to create a new driver.
+
+The components are:
+
+- tests: source code, native AST, and UAST files that with for
```
Comment (abeaumont) [inline driver/sdk.md]: s/tests: source/Tests: tests/

something is wrong with *that with for*

File: driver/sdk.md [hunk 25]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+
+A driver has several subcomponents. On this document and the next one
+we'll see how to implement all these parts to create a new driver.
+
+The components are:
+
+- tests: source code, native AST, and UAST files that with for 
+  the integration tests.
+
+- Native parser: parses source code in the driver's target language
```
Comment (abeaumont) [inline driver/sdk.md]: The parser's language doesn't need to be the driver's target language, as stated below in the documentation.

File: driver/sdk.md [hunk 26]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+language, a third-party library, or a custom one (in that order of preference).
+Any language can be used to implement the parser so use whatever is easier for the
+specific language you're working on.
 
-### JSON serialization
+#### Native parser operation
 
-The driver should contain a JSON serializer which is able to serialize requests
-and responses (including the AST) into a single-line JSON.
+When the driver executes the parser, the former entry point should Officially a
```
Comment (abeaumont) [inline driver/sdk.md]: I think *Officially* should be a different word here

File: driver/sdk.md [hunk 27]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+in the top level of the driver repository. This directory should contain all
+source code and support files (e.g. build system manifests) required to build it.
 
-The core functionality to be implemented in the native side is the AST parser.
-This should get the contents of a file and return an AST representation. This
-AST representation should be what the AST parser outputs as is. The parser might
-be from the standard library of the language, a third party library, or a custom
-one (in that order of preference).
+The core functionality to be implemented on the parser side is the source code
+parser.  This should get the source contents of a file and return an AST
```
Comment (abeaumont) [inline driver/sdk.md]: There are double spaces after a dot in a bunch of places.

File: driver/sdk.md [hunk 28]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
         continue
+
     try:
         write_to_stdout(response)
     except:
         exit(1)
 ```
 
-## Normalizing the AST
+In order to avoid maintainability burdens, the parser should avoid as much as
```
Comment (abeaumont) [inline driver/sdk.md]: avoid as much as possible to ... -> avoid to ... as much as possible

File: driver/sdk.md [hunk 29]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+possible to change the AST that is generated by the language library or third
+party module in use. One exception to this would be to improve the information
```
Comment (abeaumont) [inline driver/sdk.md]: s/One/An/

File: driver/sdk.md [hunk 30]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+
+### Creating the Converter and Annotator
+
+The conversion from AST to UAST is written in Go. The main files to be edited are
+`driver/normalizer/normalizer.go` and `driver/normalizer/parser.go`. The details
+are explained in the [annotations section](driver/annotations.md).
+
+### Updating the Makefile with the build instructions
+
+The Makefile must contain a target with instructions to build the
```
Comment (abeaumont) [inline driver/sdk.md]: The Makefile -> `Makefile`

File: driver/sdk.md [hunk 31]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+
+The building of the driver is done in a Docker container so you must adapt the
+Docker template `Dockerfile.build.tpl` that the `bblfsh` command created for you
+to install all the dependencies needed to build the `native` command detailed in
+the previous step.
+
+The templated dockerfile looks like this (without comments):
+
+```Dockerfile
+FROM alpine:3.5
```
Comment (abeaumont) [inline driver/sdk.md]: s/3.5/3.6/

File: driver/sdk.md [hunk 32]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+
+### Configuring the run Dockerfile
+
+The second dockerfile template that you have to edit is the one that the Babelfish
+server will use to run new instances of drivers. Like the build one it will also 
+be generated by the `bblfsh-sdk` command but with the filename `Dockerfile.tpl`
+and this initial content without comments:
+
+```Dockerfile
+FROM alpine:3.5
```
Comment (abeaumont) [inline driver/sdk.md]: s/3.5/3.6/

File: driver/sdk.md [hunk 33]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+dynamically loading libraries for compiled languages (not included in Alpine base)
+or runtimes for interpreted or compiled-to-bytecode languages.
+
+Then you will need to copy the `build` directory into `/opt/driver/bin`, so that
+both the (automatically build by the SDK) `driver` and `native` binaries will 
+rest in that directory.
 
-## Testing
+The final `CMD` command should remain at the end of the file and you should not
+change it.
```
Comment (abeaumont) [inline driver/sdk.md]: It'd be nice what the purpose of this `CMD` is, so that people understand why it should remain unchanged.

File: driver/sdk.md [hunk 34]
```diff
@@ -76,78 +84,269 @@ $ bblfsh-sdk update
...
+failed integration tests.
+
+### Officially adding your driver to Babelfish
+
+The last step if to have your language added to the [Babelfish Github's
+Project](https://github.com/bblfsh) so more people can use/break/debug or improve
+it.  For this you need to request a new repository to be created which you can
+make by opening a PR or talking with us in the [public Babelfish Slack
+channel](https://sourced-community.slack.com/join/shared_invite/MTkwNTM0ODEyODIzLTE0OTYxMzc5NTMtODRhMDYyNzAyYQ).
+Once the driver subproject has been added you'll make a PR with your code.
```
Comment (abeaumont) [inline driver/sdk.md]: It'd be nice to describe what *officially adding your driver to babelfish* imply. Who becomes the maintainer of the project, bblfsh organization? Will the author be made part of this organization or will they still have to add PR with any further changes? Is there any requirement before applying for *officiality*?

File: driver/annotations.md [hunk 35]
```diff
@@ -0,0 +1,171 @@
+# Adding UAST Annotations
+
+# Intro
+
+Once you have written the [AST parser](driver/sdk.md) the next step is to write
```
Comment (juanjux) [inline driver/annotations.md]: Maybe, the file is certainly long for the average page size on the book.

File: driver/annotations.md [hunk 36]
```diff
@@ -0,0 +1,171 @@
...
+the annotation code. This is the Go language code that will establish the rules to
+transform the original AST into the UAST and the implementation uses two types,
```
Comment (juanjux) [inline driver/annotations.md]: Added "a process we call normalizing" so the user (that is learning) start to learn our terminology.

File: driver/annotations.md (hunk 16 above)
Comment (juanjux) [inline driver/annotations.md]: They need to use them, tough. (applied the other fixes).

File: driver/annotations.md (hunk 18 above)
Comment (juanjux) [inline driver/annotations.md]: Changed the order. I think that the shortcut explanation is educative with respect to `HasProperty` so I'll leave it, as is in the godoc, if other reviewers doesn't see a problem.

File: driver/annotations.md (hunk 19 above)
Comment (juanjux) [inline driver/annotations.md]: ? This is the function signature and follows Go coding conventions.

File: driver/annotations.md (hunk 20 above)
Comment (juanjux) [inline driver/annotations.md]: This is after the text about asking in the Slack channel and will be also after the doc about the table @alcortesm is doing so it's really the last option, but I'll add a sentence about opening an issue as alternative to the PR. Once the BIPs are merged I would link those too.

File: driver/sdk.md (hunk 22 above)
Comment (juanjux) [inline driver/sdk.md]: I don't find it confusing. Fixed the *on*.

File: driver/sdk.md (hunk 23 above)
Comment (juanjux) [inline driver/sdk.md]: Both are fine (checked with a native) just like "algún problema" and "cualquier problema" in spanish, but *any* is more common in this sentence, so fixed.

File: driver/sdk.md (hunk 25 above)
Comment (juanjux) [inline driver/sdk.md]: The source code is (obviously) in the driver target language.

File: driver/sdk.md (hunk 28 above)
Comment (juanjux) [inline driver/sdk.md]: Native speaker: "when the sentence is long like in this case it sounds better before than after, but add commas" (added).

File: driver/sdk.md (hunk 29 above)
Comment (juanjux) [inline driver/sdk.md]: Both are fine.

File: driver/sdk.md (hunk 33 above)
Comment (juanjux) [inline driver/sdk.md]: I think it's pretty obvious, but fixed.

File: driver/sdk.md (hunk 34 above)
Comment (juanjux) [inline driver/sdk.md]: Your doubts are not defined yet, but if somebody wants to add a driver tomorrow the explained method would be the method they would have to  follow, right now.

Usually making a PR to an open source project doesn't make you automatically part of the organization or admin of the project so I don't think we need to explain that.

File: driver/sdk.md (hunk 25 above)
Comment (abeaumont) [inline driver/sdk.md]: oh, that's right :D

File: driver/sdk.md (hunk 34 above)
Comment (abeaumont) [inline driver/sdk.md]: I think we should define what an author should expect from babelfish before asking them to follow this process.

> Usually making a PR to an open source project doesn't make you automatically part of the organization or admin of the project so I don't think we need to explain that.

That's correct. It usually doesn't involve asking a project to create a new repo for your project, either, so it's a more involving process that just *making a PR*, it involves project ownership transfer.

File: driver/annotations.md (hunk 19 above)
Comment (abeaumont) [inline driver/annotations.md]: I'm talking about function definition, not its signature. The tense comment above also applies. So `adds` instead of `Will add`

Comment (juanjux) [inline driver/annotations.md]: Got it. Fixed.

File: driver/sdk.md [hunk 37]
```diff
@@ -43,7 +47,7 @@ exist, like the README.md file in the example below.
 ```bash
 $ bblfsh-sdk init mylang alpine
 initializing driver "mylang", creating new manifest
-creating file "manifest.toml"
+aftertoml"
```
Comment (smola) [inline driver/sdk.md]: this looks like a typo

File: driver/sdk.md [hunk 38]
```diff
@@ -1,39 +1,43 @@
...
 ```bash
 $ go get github.com/bblfsh/sdk/...
 ```
 
-This will fetch the sdk repository to `$GOPATH/src/github.com/bblfsh/sdk` and
+This will fetch the SDK repository to `$GOPATH/src/github.com/bblfsh/sdk` and
 will install the `bblfsh-sdk` CLI to `$GOPATH/bin/bblfsh-sdk`.
 
-## Creating a driver for a new language
+### Creating the driver's filesystem tree
```
Comment (smola) [inline driver/sdk.md]: Maybe `Creating the driver initial structure`?

**Comment by juanjux** [comment]
Feedback incorporated and rebased.

**Comment by smola** [comment]
Merging, let's iterate after this.

**Comment by juanjux** [comment]
Is there something missing from the reviews? I think I fixed/incorporated feedback from all the comments.
